src/swarm-engine/Swarm/Game/Step/Util/Command.hs

```diff
@@ -99,19 +99,19 @@ ensureCanExecute ::
   Const ->
   m ()
 ensureCanExecute c =
-  gets @Robot (constCapsFor c) >>= \case
-    Nothing -> pure ()
-    Just cap -> do
-      isPrivileged <- isPrivilegedBot
-      -- Privileged robots can execute commands regardless
-      -- of equipped devices, and without expending
-      -- a capability's exercise cost.
-      unless isPrivileged $ do
-        robotCaps <- use robotCapabilities
-        let capProviders = M.lookup cap $ getMap robotCaps
-        case capProviders of
-          Nothing -> throwError $ Incapable FixByEquip (R.singletonCap cap) (TConst c)
-          Just rawCosts -> payExerciseCost c rawCosts
+  gets @Robot (constCapsFor c) >>= mapM_ go
```

Comment:
Out of curiosity, does it work to write `gets @Robot (constCapsFor c) >>= mapM_ \cap -> do ...` ?  Either is fine though.

Reply:
Hmm, that's odd.  Oh well.